src/vs/workbench/contrib/extensions/browser/extensions.contribution.ts

```diff
@@ -257,6 +273,39 @@ CommandsRegistry.registerCommand({
 	}
 });
 
+CommandsRegistry.registerCommand({
+	id: EXTENSIONS_INSTALLVSIX_COMMAND_ID,
+	handler: async (accessor: ServicesAccessor, resources: URI[] | URI) => {
+		const extensions = Array.isArray(resources) ? resources : [resources];
```

Comment:
Looks good, from what I see this is the same code as the Action code from before thus not going into too much detail.